akka-http-core/src/main/scala/akka/http/engine/ws/FrameEventParser.scala

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.engine.ws
+
+import akka.http.util.{ ByteReader, ByteStringParserStage }
+import akka.stream.stage.{ StageState, SyncDirective, Context }
+import akka.util.ByteString
+
+import scala.annotation.tailrec
+
+/**
+ * Streaming parser for the Websocket framing protocol as defined in RFC6455
+ *
+ * http://tools.ietf.org/html/rfc6455
+ *
+ * 0                   1                   2                   3
+ * 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+ * +-+-+-+-+-------+-+-------------+-------------------------------+
+ * |F|R|R|R| opcode|M| Payload len |    Extended payload length    |
+ * |I|S|S|S|  (4)  |A|     (7)     |             (16/64)           |
+ * |N|V|V|V|       |S|             |   (if payload len==126/127)   |
+ * | |1|2|3|       |K|             |                               |
+ * +-+-+-+-+-------+-+-------------+ - - - - - - - - - - - - - - - +
+ * |     Extended payload length continued, if payload len == 127  |
+ * + - - - - - - - - - - - - - - - +-------------------------------+
+ * |                               |Masking-key, if MASK set to 1  |
+ * +-------------------------------+-------------------------------+
+ * | Masking-key (continued)       |          Payload Data         |
+ * +-------------------------------- - - - - - - - - - - - - - - - +
+ * :                     Payload Data continued ...                :
+ * + - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - +
+ * |                     Payload Data continued ...                |
+ * +---------------------------------------------------------------+
+ *
+ */
+class FrameEventParser extends ByteStringParserStage[FramePart] {
+  protected def onTruncation(ctx: Context[FramePart]): SyncDirective =
+    ctx.fail(new RuntimeException("Data truncated"))
```

Comment:
Yes, thanks, I will change that to ProtocolException which will also properly handled downstream.